### PR TITLE
(Event Sourced CR) FEATURE: Handle references between nodes in event sourced cr

### PR DESCRIPTION
**Pull request by mficzel**
- Import `reference` and `references` properties as `NodeReferencesWereSet` event
- Project `NodeReferencesWereSet` event in graph-projection to table `neos_contentgraph_referencerelation`
- Add `findReferencedNodes` to content subgraph interface and implementation
- Add fallbacks for reading of `reference` and `references` properties via getProperty or getProperties (by extending the Content/PropertyCollection)
- Add fallbacks for writing of `reference` and `references` properties setProperty

### CODE REVIEW

File: Neos.ContentRepository/Classes/Domain/Context/Node/Command/CreateReferenceBetweenNodes.php [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+use Neos\ContentRepository\Domain\ValueObject\ContentStreamIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\DimensionSpacePointSet;
+use Neos\ContentRepository\Domain\ValueObject\NodeAggregateIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\ReferencePosition;
+use Neos\ContentRepository\Domain\ValueObject\PropertyName;
+
+/**
+ * Reference a node
+ *
+ * Moves a node before, into or after another node.
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/Context/Node/Command/CreateReferenceBetweenNodes.php]: Copied comment?

File: Neos.ContentRepository/Classes/Domain/Context/Node/Event/ReferenceBetweenNodesWasCreated.php [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+namespace Neos\ContentRepository\Domain\Context\Node\Event;
+
+use Neos\ContentRepository\Domain\ValueObject\ContentStreamIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\DimensionSpacePointSet;
+use Neos\ContentRepository\Domain\ValueObject\NodeAggregateIdentifier;
+use Neos\ContentRepository\Domain\ValueObject\PropertyName;
+use Neos\EventSourcing\Event\EventInterface;
+
+/**
+ * Node was moved after, into or before another node event
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/Context/Node/Event/ReferenceBetweenNodesWasCreated.php]: Copied comment?

File: Neos.ContentRepository/Classes/Domain/Context/Node/NodeCommandHandler.php [hunk 3]
```diff
@@ -612,7 +614,41 @@ private function nodeInAggregateWasTranslatedFromCommand(TranslateNodeInAggregat
 
         return $events;
     }
+    /**
+     * @param CreateReferenceBetweenNodes $command
+     * @throws Exception\NodeException
+     */
+    public function handleCreateReferenceBetweenNodes(CreateReferenceBetweenNodes $command): void
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/Context/Node/NodeCommandHandler.php]: Really a referenceReference?

File: Neos.ContentRepository/Classes/Domain/ValueObject/PropertyName.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?php
+
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/ValueObject/PropertyName.php]: There should be no newlines before the namespace statement

File: Neos.ContentRepository/Classes/Domain/Context/Node/NodeCommandHandler.php [hunk 5]
```diff
@@ -612,7 +614,41 @@ private function nodeInAggregateWasTranslatedFromCommand(TranslateNodeInAggregat
...
+
+            $events = $this->referenceReferenceBetweenNodesWasCreated($command);
+            $this->nodeEventPublisher->publishMany(
+                ContentStreamCommandHandler::getStreamNameForContentStream($contentStreamIdentifier),
+                $events
+            );
+        });
+    }
 
+    private function referenceReferenceBetweenNodesWasCreated(CreateReferenceBetweenNodes $command): array
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/Context/Node/NodeCommandHandler.php]: Really a referenceReference?

**Comment by mficzel** [comment]
@daniellienert adjusted the code and marked as wip for now since the reference projections are still missing

**Pull request merged**